farmdata2_modules/fd2_tabs/resources/RegexInputComponent.spec.comp.js

```diff
@@ -0,0 +1,228 @@
+import { mount } from '@cypress/vue'
+import { shallowMount } from '@vue/test-utils'
+
+var RegexInputComp = require("./RegexInputComponent.js")
+var RegexInputComponent = RegexInputComp.RegexInputComponent
+
+describe('RegexInput Component', () => {
+
+        context('normal feature tests', () => {
+            beforeEach(() => {
+                mount(RegexInputComponent, {
+                    propsData: {
+                        regExp: null,
+                    }
+                })
+            })
+
+            it('exists', () => {
+                cy.get('[data-cy=regex-input]').should('exist')
+            })
+
+            it('can be typed in', () => {
+                cy.get('[data-cy=text-input]')
+                    .clear()
+                    .type('Hello World!')
+                    .should('have.value', 'Hello World!')
+        })
+
+            it('emits the new isMatch value after blur', () => {
+                const spy = cy.spy()
+                Cypress.vue.$on('is-match-changed', spy)
+                cy.get('[data-cy=text-input]')
+                    .type('Hello world!')
+                    .blur()
+                    .then(() => {
+                        expect(spy).to.be.called
+                    })
+    })
+})
+
+    context('testing disabled option', () => {
+        let comp;
+        beforeEach(() => {
```

Comment:
Should this be in before each?  If you shallow mount before each and then mount in the "it" below isn't the component mounted twice?  Once shallow and once normal?  Maybe you don't need the before each here and just mount or shallow mount in the it?  Would that work?

Reply:
When you remove the _before()_ from the context, the following test stops working: 

![image](https://user-images.githubusercontent.com/31524934/175112686-82fe26e7-747d-4cf0-9e50-586968607ae2.png)

I'm guessing when it isn't shallow mounted before each test, the test that changes the data value via the prop stops working because you need a shallow prop in order to change prop values which will trigger a watch that will change the data value.